### PR TITLE
Infinite loading of photo list

### LINKED ISSUE #2: Infinite scroll for photos in album detail

### LINKED ISSUE #41: Infinite scrolling on PhotoList

### PR DESCRIPTION
**Pull request by natalieacevedo**
Add infinite loading to PhotoList component using the intersectionObserver API.

Fixes #2.

### CODE REVIEW

File: src/components/PhotoList.tsx [hunk 1]
```diff
@@ -57,15 +81,15 @@ const PhotoList = ({ albumId, data, error, isError, isLoading }: PhotoListParams
               <Link to={photoUrl(photo.id, albumId)}>
                 {renderDefaultPhoto({ wrapped: true })}
 
-                <div className="picture-info">
+                <div className="picture-info" ref={photo.isLast ? setObserver : null}>
                   {photo.title && <h4 className="picture-title"> {photo.title}</h4>}
-                  <h5 className="picture-date">{photo.date}</h5>
+                  <h5 className="picture-dat?random=2e">{photo.date}</h5>
```
Comment (haplo) [inline src/components/PhotoList.tsx]: Looks like `?random=2` was inserted by mistake here.

File: src/hooks/useInfiniteScroll.jsx
Comment (haplo) [inline src/hooks/useInfiniteScroll.jsx:1]: Let's remove this file for now. We can later refactor into a reusable hook if we see the need.

File: src/pages/AlbumDetailPage.tsx [hunk 2]
```diff
@@ -15,14 +15,18 @@ function AlbumDetailPage() {
 
   const singleAlbumQuery = albumQuery(albumId)
   const childrenAlbumsQuery = albumsQuery({ parent: albumId, limit: 100 })
-  const photos = photosQuery({ album: albumId, limit: 50 })
+  const photos = photosQuery({
+    album: albumId,
+    limit: 20,
```
Comment (haplo) [inline src/pages/AlbumDetailPage.tsx]: Let's put a limit of `50` in desktop and `20` in mobile.

File: src/pages/PhotoListPage.tsx [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
 import PhotoList from "../components/PhotoList"
 import useApi from "../data"
 
 const PhotoListPage = () => {
   const { photosQuery } = useApi()
-  const photos = photosQuery({ limit: 100 })
-  return <PhotoList {...photos} />
+
+  const photos = photosQuery({
+    limit: 20,
```
Comment (haplo) [inline src/pages/PhotoListPage.tsx]: As in `AlbumDetailPage`: 50 for desktop, 20 for mobile.

**Review by haplo — CHANGES_REQUESTED** [review]

File: src/components/PhotoList.tsx (hunk 1 above)
Comment (natalieacevedo) [reply src/components/PhotoList.tsx]: Yes it was!

File: src/pages/AlbumDetailPage.tsx [hunk 4]
```diff
@@ -12,13 +13,14 @@ type AlbumDetailParams = {
...
-
+  const isDesktop = useDeviceDetector()
+  const numberOfPhotos = isDesktop ? 50 : 20
   const singleAlbumQuery = albumQuery(albumId)
   const childrenAlbumsQuery = albumsQuery({ parent: albumId, limit: 100 })
   const photos = photosQuery({
     album: albumId,
     limit: 20,
-    offset: 0,
+    offset: numberOfPhotos,
```
Comment (haplo) [inline src/pages/AlbumDetailPage.tsx]: This should be `limit`, not `offset`.

File: src/pages/PhotoListPage.tsx [hunk 5]
```diff
@@ -1,13 +1,14 @@
...
+import useDeviceDetector from "../hooks/useDeviceDetector"
 const PhotoListPage = () => {
+  const isDesktop = useDeviceDetector()
   const { photosQuery } = useApi()
-
+  const numberOfPhotos = isDesktop ? 50 : 20
   const photos = photosQuery({
     limit: 20,
-    offset: 0,
+    offset: numberOfPhotos,
```
Comment (haplo) [inline src/pages/PhotoListPage.tsx]: `limit`

File: src/pages/AlbumDetailPage.tsx (hunk 4 above)
Comment (natalieacevedo) [reply src/pages/AlbumDetailPage.tsx]: Sure!.. my bad!

**Review by haplo — APPROVED** [review]

**Pull request merged**